spec/unit/spotlight/dor/indexer_spec.rb

```diff
@@ -5,47 +5,204 @@
   let(:solr_doc) { {} }
+  let(:mods_loc_phys_loc) do
+    Nokogiri::XML <<-EOF
+      <mods xmlns="#{Mods::MODS_NS}">
+        <location>
+          <physicalLocation>#{example}</physicalLocation>
+        </location>
+      </mods>
+    EOF
+  end
+  let(:mods_rel_item_loc_phys_loc) do
+    Nokogiri::XML <<-EOF
+      <mods xmlns="#{Mods::MODS_NS}">
+        <relatedItem>
+          <location>
+            <physicalLocation>#{example}</physicalLocation>
+          </location>
+        </relatedItem>
+      </mods>
+    EOF
+  end
+
+  before do
+    allow(r).to receive(:harvestdor_client)
+    i = Harvestdor::Indexer.new
+    i.logger.level = Logger::WARN
+    allow(r).to receive(:indexer).and_return i
+  end
 
   describe '#add_series' do
-    let(:smods_rec) { Stanford::Mods::Record.new }
-    let(:mods_start) { "<mods xmlns='#{Mods::MODS_NS}'><location><physicalLocation>" }
-    let(:mods_end) { '</physicalLocation></location></mods>' }
-    before do
-      allow(r).to receive(:harvestdor_client)
-    end
+    # example string as key, expected series name as value
+    {
+      # feigenbaum
+      'Call Number: SC0340, Accession 2005-101': '2005-101',
+      'Call Number: SC0340, Accession 2005-101, Box : 39, Folder: 9': '2005-101',
+      'Call Number: SC0340, Accession 2005-101, Box: 2, Folder: 3': '2005-101',
+      'Call Number: SC0340, Accession: 1986-052': '1986-052',
+      'Call Number: SC0340, Accession: 1986-052, Box 3 Folder 38': '1986-052',
+      'Call Number: SC0340, Accession: 2005-101, Box : 50, Folder: 31': '2005-101',
+      'Call Number: SC0340, Accession: 1986-052, Box: 5, Folder: 1': '1986-052',
+      'SC0340, Accession 1986-052': '1986-052',
+      'SC0340, Accession 2005-101, Box 18': '2005-101',
+      'Call Number: SC0340, Accession 2005-101, Box: 42A, Folder: 24': '2005-101',
+      'Call Number: SC0340, Accession: 1986-052, Box: 42A, Folder: 59': '1986-052',
+      'SC0340': nil,
+      'SC0340, 1986-052, Box 18': nil,
+      'Stanford University. Libraries. Department of Special Collections and University Archives': nil,
+      # shpc (actually in <relatedItem><location><physicalLocation>)
+      'Series Biographical Photographs | Box 42 | Folder Abbot, Nathan': 'Biographical Photographs',
+      'Series General Photographs | Box 42 | Folder Administration building--Outer Quad': 'General Photographs',
+      # menuez
+      'MSS Photo 451, Series 1, Box 32, Folder 11, Sleeve 32-11-2, Frame B32-F11-S2-6': '1',
+      'Series 1, Box 10, Folder 8': '1',
+      # fuller
+      'Collection: M1090 , Series: 4 , Box: 5 , Folder: 10': '4',
+      # hummel (actually in <relatedItem><location><physicalLocation>)
+      'Box 42 | Folder 3': nil,
+      'Flat-box 228 | Volume 1': nil
+    }.each do |example, expected|
+      describe "for example '#{example}'" do
+        let(:example) { example }
+        context 'in /location/physicalLocation' do
+          before do
+            i = Harvestdor::Indexer.new
```

Comment:
I'll try it without.  I think I did it to quiet the annoying messages in the tests, but i may have done it in two places.